doc/tutorials/highlevel.md

```diff
@@ -0,0 +1,322 @@
+example to access the elements of the array `myfloatarray/#`, we obviously need to provide an index.
+
+Next we see some local helper macros only used in this file.
+
+Then the main part of the header: For each config value we generate an `ELEKTRA_GET(*)` and an `ELEKTRA_SET(*)` accessor function. All these
+functions are `static inline`, because they just call other getter/setter functions with partially fixed arguments. In fact many of these
+functions will only be a single line.
+
+Then we undefine the local macros we defined before and declare the three initialization functions `loadConfiguration`, `printHelpMessage`
+and `specloadCheck`.
+
+At the end of the file you will find the `elektra*` convenience macros. These macros can be used to make accessing config values look more
+like normal function calls and avoid the ugly double parentheses in e.g. `ELEKTRA_GET (...) (...)`.
+
+### Obtaining an `Elektra` handle
+
+We start at the bottom of our `conf.h` excerpt. `specloadCheck` is used to initiate specload mode, if needed. This mode makes your application
+provide its specification to Elektra. How this works exactly is not so important (see [specload plugin](/src/plugins/specload/README.md)).
+You only need to know, that `specloadCheck` should be called immediately at the start of your `main` function and that it only returns, when
+your application is not in specload mode.
+
+```c
+int main (int argc, const char ** argv) {
+    specloadCheck (argc, argv);
+    // ...
+}
+```
+
+To access your configuration, you first need to call `loadConfiguration` to get an `Elektra` handle for your application. This is done via
+a snippet that is more or less the same for all applications:
+
+```c
+ElektraError * error = NULL;
+Elektra * elektra = NULL;
+int rc = loadConfiguration (&elektra, &error);
+
+if (rc == -1)
+{
+    fprintf (stderr, "An error occurred while opening elektra: %s", elektraErrorDescription (error));
+    elektraErrorReset (&error);
+    exit (EXIT_FAILURE);
+}
+
+if (rc == 1)
+{
+    // help mode - application was called with '-h' or '--help'
+    printHelpMessage (elektra, NULL, NULL);
+    elektraClose (elektra);
+    exit (EXIT_SUCCESS);
+}
+```
+
+Next it is recommended, you change the default handler for fatal errors. By default we just call `exit (EXIT_FAILURE)`, since we don't know
+how you log your errors and what cleanup may be needed.
+
+```c
+elektraFatalErrorHandler (elektra, onFatalError);
+```
+
+`onFatalError` will receive the fatal `ElektraError *`. It must at least call `elektraErrorReset` on the error and then call `exit()`.
+
+### Reading config values
+
+Once you have your `Elektra` instance, reading config values is easy. You just call one of the getter functions.
+
+```c
+kdb_double_t mydouble = ELEKTRA_GET (Mydouble) (elektra);
+```
+
+Alternatively, you can use the convenience macro `elektraGet` and/or the tag macros. We recommend you use one of the ways below, to avoid
```

Comment:
I will remove `ELEKTRA_GET (Mydouble) (...)`, if you want to (although it explains what the tag macros actually do).

I will not remove `ELEKTRA_GET (ELEKTRA_TAG_MYDOUBLE) (...)`. Yes it is more verbose than `elektraGet (...)`, you cannot get a function pointer via the `elektraGet` macro.

Reply:
It is okay to (at one point) explain how it works. But here we already switched to the user-view. And the user of the API should not use `ELEKTRA_GET (Mydouble) (elektra)`.

Reply:
Btw. Why would you need a function pointer? Aren't such situations better covered with the internalnotification?

Reply:
> the user of the API should not use `ELEKTRA_GET (Mydouble) (elektra)`

Like I said, I will remove  `ELEKTRA_GET (Mydouble) (elektra)`.
But I see no technical reason to discourage  `ELEKTRA_GET (ELEKTRA_TAG_MYDOUBLE) (elektra)`. Difference between `ELEKTRA_GET (ELEKTRA_TAG_MYDOUBLE) (elektra)` and `elektraGet (elektra, ELEKTRA_TAG_MYDOUBLE)` is purely personal preference.

> Why would you need a function pointer?

I don't know, but I'm sure somebody somewhere has a use case. Mostly I don't see a reason to hide `ELEKTRA_GET (...) (...)` in the documentation. It is immediately obvious from looking at either the definition of `elektraGet` or the signature of any of the `static inline` functions that this will work. I also dislike APIs/libraries that pretend to be magic, when in fact they are not complicated at all and could easily explain how everything works.

> Aren't such situations better covered with the internalnotification?

Depends what you do with the function pointer. Also the internalnotification stuff is not supported via the highlevel API right now.

Reply:
> But I see no technical reason to discourage [...] Mostly I don't see a reason to hide ELEKTRA_GET (...) (...) in the documentation

It is only about if we really want to maintain two different kind of APIs in one API.

> I also dislike APIs/libraries that pretend to be magic, when in fact they are not complicated at all and could easily explain how everything works.

You are completely right here. It is very good that you explain. But it is something different if we also give guarantees about API stability.

> Depends what you do with the function pointer. Also the internalnotification stuff is not supported via the highlevel API right now.

Hopefully they can coexist peacefully in the same application. It is not tested, though.